plugin/pkg/scheduler/factory/factory.go

```diff
@@ -531,12 +531,20 @@ func (factory *ConfigFactory) makeDefaultErrorFunc(backoff *podBackoff, podQueue
 			}
 			// Get the pod again; it may have changed/been scheduled already.
 			pod = &api.Pod{}
-			err := factory.Client.Get().Namespace(podID.Namespace).Resource("pods").Name(podID.Name).Do().Into(pod)
-			if err != nil {
-				if !errors.IsNotFound(err) {
-					glog.Errorf("Error getting pod %v for retry: %v; abandoning", podID, err)
+			getBackoff := time.Duration(100 * time.Millisecond)
+			for {
+				if err := factory.Client.Get().Namespace(podID.Namespace).Resource("pods").Name(podID.Name).Do().Into(pod); err == nil {
+					break
 				}
-				return
+				if errors.IsNotFound(err) {
+					glog.Warning("A pod %v no longer exists", podID)
+					return
+				}
+				glog.Errorf("Error getting pod %v for retry: %v; retrying...", podID)
+				if getBackoff = getBackoff * 2; getBackoff > time.Minute {
```

Comment:
Right, sorry - but you'll still multiply it all the time, even when there's no need for it.